### PR TITLE
Convert Keymaps to use STAILQ

### PR DESCRIPTION
**Pull request by yousefakbar**
* **What does this PR do?**
Converts Keymaps and its functions to use STAILQ structures instead of being a simple array of linked lists.

* **What are the relevant issue numbers?**
Issue #1179

### CODE REVIEW

File: keymap.c [hunk 1]
```diff
@@ -195,6 +195,38 @@ static const struct Extkey ExtKeys[] = {
...
+{
+  struct Keymap *np = NULL, *tmp = NULL;
+
+  STAILQ_FOREACH_SAFE(np, km_list, entries, tmp)
+  {
+    STAILQ_REMOVE(km_list, np, Keymap, entries);
+    mutt_keymap_free(&np);
+  }
+
+  STAILQ_INIT(km_list);
```
Comment (gahr) [inline keymap.c:227]: So I think you're mixing two approaches here:

1. iterate, remove and free
```c
struct Keymap *np = NULL, *tmp = NULL;
STAILQ_FOREACH_SAFE(np, km_list, entries, tmp)
{
  STAILQ_REMOVE(km_list, np, Keymap, entries)
  mutt_keymap_free(&np);
}
```

and

2. iterate, free, adjust pointers at the end
```c
struct Keymap *np = STAILQ_FIRST(km_list), *next = NULL;
while (np)
{
  next = STAILQ_NEXT(np, entries);
  mutt_keymap_free(&np);
}
STAILQ_INIT(km_list);
```

By looking for `STAILQ_INIT`  in the code, it seems you're not the first one :)

**Comment by yousefakbar** [comment]
Misclicked something in review on my phone. Is there something I need to change in my commits? My view is limited until I'm back on keyboard.

**Comment by gahr** [comment]
I think the code looks ok - please have a look at my comment from yesterday regarding alternative ways to clean up a STAILQ. Please pick one - your code is using a mix of the two.

File: keymap.c [hunk 2]
```diff
@@ -347,82 +401,73 @@ static enum CommandResult km_bind_err(const char *s, enum MenuType menu, int op,
...
+        char old_binding[MAX_SEQ];
+        char new_binding[MAX_SEQ];
+        km_expand_key(old_binding, MAX_SEQ, map);
+        km_expand_key(new_binding, MAX_SEQ, np);
+        if (err)
+        {
+          /* err was passed, put the string there */
+          snprintf(
+              err->data, err->dsize,
+              _("Binding '%s' will alias '%s'  Before, try: 'bind %s %s "
```
Comment (gahr) [inline keymap.c:437]: There are two calls to gettext with identical parameters. Can you please save the result in a variable and use it in both places?

File: keymap.c (hunk 1 above)
Comment (yousefakbar) [reply keymap.c]: Fixed. Went with option 1.

File: keymap.c (hunk 2 above)
Comment (yousefakbar) [reply keymap.c]: Done. Stored result in a string called 'err_msg'.

**Comment by flatcap** [comment]
I've made a few minor changes:
- rebased
- fixed Doxygen comments
- run clang-format
- tidied commit message

**Pull request merged**